sdk/core/System.ClientModel/src/Convenience/StreamingClientResultOfT.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class StreamingClientResult<T> : ClientResult, IAsyncEnumerable<T>
```

Comment:
The only thing this does is abstractly implement the `IAsyncEnumerable<T>` interface, and provide a protected constructor that calls through to the base constructor.

We still have an open question of whether this should implement IDisposable to make it easy to dispose the response content stream by putting this result in a using block, but I'm interested to first investigate how hard it is to use the IAsyncDisposable interface implementation in the IAsyncEnumerator returned from GetAsyncEnumerator.